packages/react/src/auto/polaris/PolarisDateTimePicker.tsx

```diff
@@ -48,58 +14,120 @@ const copyTime = (to: Date, input: Date) => {
   to.setMilliseconds(input.getMilliseconds());
 };
 
+const createMarkup = (items: string[], selectedState: string) => {
+  return items.map((item, i) => {
+    return (
+      <div key={i} style={{ textAlign: "center", cursor: "default", padding: "0px 4px" }}>
+        <Listbox.Option value={item}>
+          {selectedState.padStart(2, "0") === item ? (
+            <Box padding="100" background="bg-surface-secondary-selected" borderRadius="200" minHeight="30px" minWidth="30px">
+              {item}
+            </Box>
+          ) : (
+            <Box minHeight="30px" minWidth="30px" padding="100">
+              {item}
+            </Box>
+          )}
+        </Listbox.Option>
+      </div>
+    );
+  });
+};
+
+const hours = Array.from({ length: 12 }, (_, i) => (i + 1).toString().padStart(2, "0"));
+const mins = Array.from({ length: 60 }, (_, i) => i.toString().padStart(2, "0"));
+const ampm = ["AM", "PM"];
+
+const timeFormatRegex = new RegExp(/((1[0-2]|0?[1-9]):([0-5][0-9]) ?([AaPp][Mm]))/);
 const today = new Date();
 export const PolarisDateTimePicker = (props: {
+  field: string;
   id?: string;
   value?: Date;
-  includeTime?: boolean;
-  dateLabel?: string;
   timeLabel?: string;
-  onChange: (value: Date) => void;
+  onChange?: (value: Date) => void;
   error?: string;
+  includeTime?: boolean;
+  hideTimePopover?: boolean;
 }) => {
+  const { path, metadata } = useFieldMetadata(props.field ?? "");
+
+  const { field: fieldProps } = useController({
+    name: path,
+  });
+
   const { onChange, value } = props;
   const localTz = Intl.DateTimeFormat().resolvedOptions().timeZone;
-  const localTime = value ? utcToZonedTime(value, localTz) : undefined;
+  const localTime = value ? utcToZonedTime(value, localTz) : fieldProps.value;
   const [{ month, year }, setDate] = useState({ month: (localTime ?? today).getMonth(), year: (localTime ?? today).getFullYear() });
   const [datePopoverActive, setDatePopoverActive] = useState(false);
+  const [timeString, setTimeString] = useState(localTime ? format(localTime, "K:m aa") : "");
+  const [popoverActive, setPopoverActive] = useState(false);
+  const [timeParseError, setTimeParseError] = useState(false);
+  const [hourSelected, setHourSelected] = useState(hours[0]);
+  const [minSelected, setMinSelected] = useState(mins[0]);
+  const [ampmSelected, setAmpmSelected] = useState(ampm[0]);
```

Comment:
Can these be changed such that they are parsed from `timeString`? That would prevent the possibility of these values disagreeing with `timeString` which is the real source of truth.  

It could be nice to have a single object represent the whole dateTime to prevent string operations too. Maybe something like this

```js
const  [dateTimeState, setDateTimeState] = useState({
minute,
hour,
ampm,
day,
month,
year

})

const setHours = (hour:number) => {
   setDateTimeState({...dateTimeState, hour})
}
```